proxying_test.go

```diff
@@ -0,0 +1,52 @@
+package fronted
+
+import (
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/getlantern/proxy"
+	"github.com/stretchr/testify/assert"
+)
+
+const (
+	humansText = "Google is built by a large team of engineers, designers, researchers, robots, and others in many different sites across the globe. It is updated continuously, and built with more tools and technologies than we can shake a stick at. If you'd like to help us out, see google.com/careers.\n"
+)
+
+func TestProxying(t *testing.T) {
+	dir, err := ioutil.TempDir("", "direct_test")
+	if !assert.NoError(t, err, "Unable to create temp dir") {
+		return
+	}
+	defer os.RemoveAll(dir)
+	cacheFile := filepath.Join(dir, "cachefile.3")
+	ConfigureCachingForTest(t, cacheFile)
```

Comment:
What is this caching for?

Reply:
It just sets fronted with a default sets of masquerades so that we can do live testing.